File: HeartForCharity/HeartForCharity.Services/Database/Address.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HeartForCharity.Services.Database
{
    public class Address
    {
        [Key]
        public int AddressId { get; set; }

        [MaxLength(200)]
        public string? StreetName { get; set; }

        [MaxLength(20)]
        public string? Number { get; set; }

        [MaxLength(20)]
        public string? PostalCode { get; set; }

        [Required]
        [ForeignKey(nameof(City))]
        public int CityId { get; set; }

        public virtual City City { get; set; } = null!;
        public virtual ICollection<UserProfile> UserProfiles { get; set; } = new List<UserProfile>();
        public virtual ICollection<OrganisationProfile> OrganisationProfiles { get; set; } = new List<OrganisationProfile>();
        public virtual ICollection<VolunteerJob> VolunteerJobs { get; set; } = new List<VolunteerJob>();
    }
}

File: HeartForCharity/UI/heartforcharity_mobile/lib/model/responses/address.dart
class Address {
  final int addressId;
  final String? streetName;
  final String? number;
  final String? postalCode;
  final int cityId;
  final String cityName;
  final String countryName;

  Address({
    this.addressId = 0,
    this.streetName,
    this.number,
    this.postalCode,
    this.cityId = 0,
    this.cityName = '',
    this.countryName = '',
  });

  factory Address.fromJson(Map<String, dynamic> json) => Address(
        addressId: json['addressId'] ?? 0,
        streetName: json['streetName'],
        number: json['number'],
        postalCode: json['postalCode'],
        cityId: json['cityId'] ?? 0,
        cityName: json['cityName'] ?? '',
        countryName: json['countryName'] ?? '',
      );
}
